### PR TITLE
[WIP] Rust implementation

### PR DESCRIPTION
**Pull request by garritfra**
This might come out of the blue: I've tried rewriting a tiny function in rust to see what its like, and if it works. And sure enough, it works like a charm! I was able to implement the function `isSupportedNodeVersion` in rust. I chose this function, because it does not have a vital impact, yet its easy to implement. Please let me know what you think of this.

### TODO

- [x] Fix Lint
- [ ] Figure out way to publish binaries
- [x] Remove old utils module
- [ ] Find way to recompile after changes
- [ ] Add compile instructions to README

### CODE REVIEW

**Comment by garritfra** [comment]
**Mental note**: Linting fails because nested node modules are not ignored by ESLint

File: cli/index.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
 #!/usr/bin/env node
 
 const chalk = require("chalk");
-const { isSupportedNodeVersion } = require("../utils/versionUtils");
+
+// TODO: How to link lerna with unpublished package?
```
Comment (garritfra) [inline cli/index.js]: @andreyluiz any idea how to fix this?

**Comment by garritfra** [comment]
@andreyluiz any thoughts how to fix the thing I mentioned in my review?

File: cli/index.js (hunk 1 above)
Comment (andreyluiz) [reply cli/index.js]: You need to add the dependency for the `clio-utils@0.1.0` package on the clio project `package.json`. Npm will automagically use lerna to know where to get the package.

File: packages/utils/native/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[package]
+name = "utils"
```
Comment (andreyluiz) [inline packages/utils/native/Cargo.toml]:
```suggestion
name = "clio-utils"
```

Comment (garritfra) [reply packages/utils/native/Cargo.toml]: Didn't work either🧐

**Comment by garritfra** [comment]
Evaluation concluded that Rust isn't really suited for node modules. We would either have to ship compiled binaries for every platform, host these binaries somewhere and download them on install time or compile it during installation of clio. None of these are valid options for us. We will however consider writing native modules in C/C++ in the future, given that virtually anyone has a corresponding compiler on their system.

**Comment by pouya-eghbali** [comment]
@garrit we can still do rust, and compile to universal wasm binaries, and ship the wasm files instead

**Comment by garritfra** [comment]
@pouya-eghbali that's probably faster than js, but it doesn't give us native performance

**Pull request closed without merging**